src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.c

```diff
@@ -317,11 +315,11 @@ static void pf_update_locked(grpc_exec_ctx *exec_ctx, grpc_lb_policy *policy,
     return;
   }
   const grpc_lb_addresses *addresses = arg->value.pointer.p;
-  size_t num_addrs = 0;
   for (size_t i = 0; i < addresses->num_addresses; i++) {
-    if (!addresses->addresses[i].is_balancer) ++num_addrs;
+    // If there were any balancer, we would have chosen grpclb policy instead.
+    GPR_ASSERT(!addresses->addresses[i].is_balancer);
   }
-  if (num_addrs == 0) {
+  if (addresses->num_addresses == 0) {
```

Comment:
You can move this `if` up, before the `for`: we don't need to count anything anymore.